testing/common/foreignkey_test.go

```diff
@@ -0,0 +1,206 @@
+	projectID = os.Getenv("HARBOURBRIDGE_TESTS_GCLOUD_PROJECT_ID")
+	instanceID = os.Getenv("HARBOURBRIDGE_TESTS_GCLOUD_INSTANCE_ID")
+
+	ctx := context.Background()
+	flag.Parse() // Needed for testing.Short().
+	noop := func() {}
+
+	if testing.Short() {
+		log.Println("Unit test for UpdateDDLForeignKeys skipped in -short mode.")
+		return noop
+	}
+
+	if projectID == "" {
+		log.Println("Unit test for UpdateDDLForeignKeys skipped: HARBOURBRIDGE_TESTS_GCLOUD_PROJECT_ID is missing")
+		return noop
+	}
+
+	if instanceID == "" {
+		log.Println("Unit test for UpdateDDLForeignKeys skipped: HARBOURBRIDGE_TESTS_GCLOUD_INSTANCE_ID is missing")
+		return noop
+	}
+
+	var err error
+	databaseAdmin, err = database.NewDatabaseAdminClient(ctx)
+	if err != nil {
+		log.Fatalf("cannot create databaseAdmin client: %v", err)
+	}
+
+	return func() {
+		databaseAdmin.Close()
+	}
+}
+
+func dropDatabase(t *testing.T, dbPath string) {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+	defer cancel()
+	// Drop the testing database.
+	if err := databaseAdmin.DropDatabase(ctx, &databasepb.DropDatabaseRequest{Database: dbPath}); err != nil {
+		t.Fatalf("failed to drop testing database %v: %v", dbPath, err)
+	}
+}
+
+func BuildConv(t *testing.T) *internal.Conv {
+	conv := internal.MakeConv()
+	conv.SpSchema["table_a"] = ddl.CreateTable{
+		Name:     "table_a",
+		ColNames: []string{"col1", "col2", "col3", "col4", "col5", "col6"},
+		ColDefs: map[string]ddl.ColumnDef{
+			"col1": ddl.ColumnDef{Name: "col1", T: ddl.Type{Name: ddl.String, Len: int64(10)}},
+			"col2": ddl.ColumnDef{Name: "col2", T: ddl.Type{Name: ddl.String, Len: int64(10)}},
+			"col3": ddl.ColumnDef{Name: "col3", T: ddl.Type{Name: ddl.String, Len: int64(10)}},
+			"col4": ddl.ColumnDef{Name: "col4", T: ddl.Type{Name: ddl.String, Len: int64(10)}},
+			"col5": ddl.ColumnDef{Name: "col5", T: ddl.Type{Name: ddl.String, Len: int64(10)}},
+			"col6": ddl.ColumnDef{Name: "col6", T: ddl.Type{Name: ddl.String, Len: int64(10)}},
+		},
+		Pks: []ddl.IndexKey{ddl.IndexKey{Col: "col1"}},
+	}
+	conv.SpSchema["table_b"] = ddl.CreateTable{
+		Name:     "table_b",
+		ColNames: []string{"col1", "col2", "col3", "col4", "col5", "col6"},
+		ColDefs: map[string]ddl.ColumnDef{
+			"col1": ddl.ColumnDef{Name: "col1", T: ddl.Type{Name: ddl.String, Len: int64(10)}},
+			"col2": ddl.ColumnDef{Name: "col2", T: ddl.Type{Name: ddl.String, Len: int64(10)}},
+			"col3": ddl.ColumnDef{Name: "col3", T: ddl.Type{Name: ddl.String, Len: int64(10)}},
+			"col4": ddl.ColumnDef{Name: "col4", T: ddl.Type{Name: ddl.String, Len: int64(10)}},
+			"col5": ddl.ColumnDef{Name: "col5", T: ddl.Type{Name: ddl.String, Len: int64(10)}},
+			"col6": ddl.ColumnDef{Name: "col6", T: ddl.Type{Name: ddl.String, Len: int64(10)}},
+		},
+		Pks: []ddl.IndexKey{ddl.IndexKey{Col: "col1"}},
+	}
+	return conv
+}
+
+func addForeignKeysToConv(conv *internal.Conv, t *testing.T) {
+	var foreignKeys []ddl.Foreignkey
+	for i := 1; i <= 6; i++ {
```

Comment:
suggest adding a few tests - single worker with multiple foreignKeys, multiple workers with a single foreignKey, number of workers more than the number of foreignKeys, number of workers equal to the number of foreignKeys, number of workers much less than the number of foreignKeys

note that you would need to refactor much of tests to be able to reuse most of the code for the same